Automattic/newspack-plugin
# Use php-loremipsum For Starter Content Text

### All Submissions:






### Changes proposed in this Pull Request:

The starter content feature generates lorem ipsum text using third party APIs. This approach is slow and failure prone. In this branch the [php-loremipsum](https://github.com/joshtronic/php-loremipsum) library is installed and used instead. 

From the php-loremipsum docs:

> The first string generated will always start with the traditional "Lorem ipsum dolor sit amet, consectetur adipiscing elit". Subsequent strings may contain those words but will not explicitly start with them.

This is problematic because it means all titles will be identical, and the point of starter content is to display a random, realistic-looking body of content. To beat this, this branch strips out the identical opening bits of generated lorem ipsum to make each string unique. In the case of paragraphs, the first one is removed which guarantees the opening will be unique.

### How to test the changes in this Pull Request:

1. Switch to this branch, run `composer update`.
2. Go to `/wp-admin/admin.php?page=newspack-setup-wizard#/starter-content`, and click Install Starter Content.
3. Verify the wizard completes successfully.
4. Verify 40 new posts are created, each with unique Latin titles.
5. Verify the posts contain 5 paragraphs of lorem ipsum, which appears to be different from post to post.

### Other information:
